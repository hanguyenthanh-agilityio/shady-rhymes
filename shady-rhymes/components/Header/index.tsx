// Components
import SearchInput from '../SearchInput';
import { Flex, VStack, Stack, Text, Heading } from '@chakra-ui/react';

const Hero = () => {
  return (
    <Flex
      w="full"
      h="1000px"
      maxH={{ xs: '550px', lg: '1000px' }}
      backgroundImage="url(/images/hero.png)"
      backgroundSize="cover"
      backgroundPosition="center center"
    >
      <VStack
        w="100%"
        justify="center"
        // bgGradient="linear(to-r, blackAlpha.600, transparent)"
      >
        <Stack align="center">
          <Heading
            size={{ xs: 'medium', md: 'huge' }}
            textAlign="center"
            maxW={{ xs: '300px', md: '800px' }}
            color="#fff"
          >
            Make your interior more minimaistic & modern
          </Heading>
          <Text
            size={{ xs: 'tiny', md: 'medium' }}
            textAlign="center"
            color="#fff"
            maxW={{ xs: '260px', md: '500px' }}
          >
            Bringing interiors to life, understanding the person of people
            living in a home is far greater need in design.
          </Text>
        </Stack>
        <Stack align="center" pt="60px" pb="100px">
          <SearchInput />
        </Stack>
      </VStack>
    </Flex>
  );
};

export default Hero;
