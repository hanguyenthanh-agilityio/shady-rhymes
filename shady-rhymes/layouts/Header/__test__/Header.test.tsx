import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { ChakraProvider } from '@chakra-ui/react';

// Components
import Header from '..';

describe('Header component', () => {
  it('Should render Header snapshot correctly', () => {
    const header = render(
      <ChakraProvider>
        <Header />
      </ChakraProvider>
    );

    expect(header).toMatchSnapshot();
  });
});
