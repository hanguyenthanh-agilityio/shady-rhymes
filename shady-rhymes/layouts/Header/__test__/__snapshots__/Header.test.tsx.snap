// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render Header snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <div
        class="css-hfom1y"
      >
        <div
          class="chakra-stack css-10jsirv"
        >
          <div
            class="chakra-stack css-owjkmg"
          >
            <h2
              class="chakra-heading css-grg82q"
            >
              Make your interior more minimaistic & modern
            </h2>
            <p
              class="chakra-text css-1o6irx4"
            >
              Bringing interiors to life, understanding the person of people living in a home is far greater need in design.
            </p>
          </div>
          <div
            class="chakra-stack css-ko33wr"
          >
            <div
              class="chakra-input__group css-1tv3gtv"
              data-group="true"
            >
              <input
                class="chakra-input css-nphdvr"
                data-testid="search-input-field"
                name="search-input"
                placeholder="Search Furniture"
              />
              <div
                class="chakra-input__right-element css-ur97uj"
              >
                <svg
                  class="chakra-icon chakra-icon css-1fyhvvq"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="css-hfom1y"
    >
      <div
        class="chakra-stack css-10jsirv"
      >
        <div
          class="chakra-stack css-owjkmg"
        >
          <h2
            class="chakra-heading css-grg82q"
          >
            Make your interior more minimaistic & modern
          </h2>
          <p
            class="chakra-text css-1o6irx4"
          >
            Bringing interiors to life, understanding the person of people living in a home is far greater need in design.
          </p>
        </div>
        <div
          class="chakra-stack css-ko33wr"
        >
          <div
            class="chakra-input__group css-1tv3gtv"
            data-group="true"
          >
            <input
              class="chakra-input css-nphdvr"
              data-testid="search-input-field"
              name="search-input"
              placeholder="Search Furniture"
            />
            <div
              class="chakra-input__right-element css-ur97uj"
            >
              <svg
                class="chakra-icon chakra-icon css-1fyhvvq"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
